### PR TITLE
Bump golangci-lint to v1.49.0

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

 

 
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
- Bump golangci-lint to 1.49.0
- Fix lint error for usage of net/http (gosec)

The following linters are deprecated in golangci-lint:

structcheck - Finds unused struct fields
deadcode - Finds unused code
varcheck - Finds unused global variables and constants
ifshort - Go linter that checks if your code uses short syntax for if-statements whenever possible. (Buggy)

Remove these linters as they're no longer maintained. (Ref -https://github.com/golangci/golangci-lint/issues/1841)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Bump golangci-lint to 1.49.0
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -258,7 +258,15 @@ func main() {
 	if profilerAddress != "" {
 		setupLog.Info("Profiler listening for requests", "profiler-address", profilerAddress)
 		go func() {
-			setupLog.Error(http.ListenAndServe(profilerAddress, nil), "listen and serve error")
+			server := &http.Server{
+				Addr:              profilerAddress,
+				ReadHeaderTimeout: 3 * time.Second,
```
Comment (mboersma) [inline main.go:263]: From the [docs](https://pkg.go.dev/net/http#Server):

> 	// ReadHeaderTimeout is the amount of time allowed to read
	// request headers. The connection's read deadline is reset
	// after reading the headers and the Handler can decide what
	// is considered too slow for the body. If ReadHeaderTimeout
	// is zero, the value of ReadTimeout is used. If both are
	// zero, there is no timeout.

So I'm guessing that means with the previous code, this value was effectively 0, meaning "no timeout"?

3 seconds seems generous to read an HTTP header, so I think this is probably a safe change.

**Review by mboersma — COMMENTED** [review]
/lgtm

**Comment by Prajyot-Parab** [comment]
```
azure/defaults_test.go:95:30: should rewrite http.NewRequestWithContext or add (*Request).WithContext (noctx)
                req, err := http.NewRequest("GET", testSrv.URL, http.NoBody)
```
Fixed above linter issue post rebase.

**Pull request merged**